Support multiple values to search/exclude (fixes #4119)

def- opened the pull request:
![screenshot-20220925@143701](https://user-images.githubusercontent.com/2335377/192143966-bd91670c-bcdd-45ea-90aa-189a9fce0975.png)
## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

Robyt3 commented:
The maximum search string (127 times the character `;`) more than halves my FPS. Only affects the search bar and not the exclude bar. After some more testing it seems that searching for just `;;` will already bring down the FPS that much and all servers are found with this search. It highlights the player count for every server but no player name in particular, so I guess it tries to match the empty string with every player name in this case. Can probably be avoided.

With a string resulting in many matches (e.g. `a;b;c;d;e;f;g;h;i;j;k;l;m;n;o;p;q;r;s;t;u;v;w;x;y;z`) I also get noticeable lag spikes when editing the string:

![gametime-margin-spikes](https://user-images.githubusercontent.com/23437060/194590160-c3de9f72-4fbe-4d8b-9b7f-fbd2c09c807d.png)

Spikes get noticeable after around 10 different search terms for me, so it's likely not a big deal for most players searching for more realistic terms.

This also only seems to highlight the first match in every name, which is noticeable if you search for the alphabet string. Not sure how required this is, as it would likely further reduce the FPS when using this feature.

def- commented:
> The maximum search string (127 times the character ;) more than halves my FPS

Fixed.

> It highlights the player count for every server but no player name in particular, so I guess it tries to match the empty string with every player name in this case. Can probably be avoided.

Fixed.

> Spikes get noticeable after around 10 different search terms for me, so it's likely not a big deal for most players searching for more realistic terms.

For me they are much smaller, not sure if it's that bad.

> This also only seems to highlight the first match in every name, which is noticeable if you search for the alphabet string. Not sure how required this is, as it would likely further reduce the FPS when using this feature.

This was intentional, I thought not worth it to keep searching. Consistent with previous behavior, if a string matched twice, only the first match was shown too.

Robyt3 commented:
Thanks. I think the lag is minor enough for normal use cases.

bors r+
